neptune/new/internal/utils/deprecation.py

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (c) 2021, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from functools import wraps
+
+from neptune.new.exceptions import NeptuneInitParametersCollision
+from neptune.new.internal.utils.logger import logger
+
+
+def deprecated_id_parameter(*, deprecated_kwarg_name):
+    def deco(f):
+        @wraps(f)
+        def inner(*args, **kwargs):
+            deprecated_param = kwargs.get(deprecated_kwarg_name)
+            with_id = kwargs.get("with_id")
+            if deprecated_param is not None:
+                if with_id is not None:
+                    raise NeptuneInitParametersCollision(
+                        "with_id", deprecated_kwarg_name, method_name=f.__name__
+                    )
+
+                logger.warning(
+                    f"parameter `{deprecated_kwarg_name}` is deprecated, use `with_id` instead."
```

Comment:
It's called argument, not parameter, isn't it?

Reply:
Why do you think parameter is not ok in this scenario?

I think we're marking parameter of the function as deprecated, so user trying to pass it as an argument can't do that.

For me two options are ok, but parameter is better.

Reply:
>Why do you think parameter is not ok in this scenario?

I don't know what scenario has to do with this.
Isn't it officially commonly called **arguments** in all programming languages including python?

Reply:
Sorry, my mistake.
declaration => parameter
value pass to function call => argument